src/matrix_validator/validator_polars.py

```diff
@@ -31,55 +47,77 @@ def validate(self, nodes_file_path, edges_file_file_path):
             validate_kg_edges(edges_file_file_path, self.output_format, self.get_report_file())
 
 
+def get_biolink_model_prefix_keys():
+    from . import resources
+
+    with il_resources.open_text(resources, "biolink-model.yaml") as file:
+        bl_model_data = list(yaml.load_all(file, Loader=SafeLoader))
+    return list(bl_model_data[0]["prefixes"].keys())
+
+
+def get_biolink_model_knowledge_level_keys():
+    from . import resources
+
+    with il_resources.open_text(resources, "biolink-model.yaml") as file:
+        bl_model_data = list(yaml.load_all(file, Loader=SafeLoader))
+    return list(bl_model_data[0]["enums"]["KnowledgeLevelEnum"]["permissible_values"].keys())
+
+
+def get_biolink_model_agent_type_keys():
+    from . import resources
+
+    with il_resources.open_text(resources, "biolink-model.yaml") as file:
+        bl_model_data = list(yaml.load_all(file, Loader=SafeLoader))
+    return list(bl_model_data[0]["enums"]["AgentTypeEnum"]["permissible_values"].keys())
+
+
 def validate_kg_nodes(nodes, output_format, report_file):
     """Validate a knowledge graph using optional nodes TSV files."""
-    validation_reports = []
-
     logger.info("Validating nodes TSV...")
 
+    bm_prefixes = get_biolink_model_prefix_keys()
+
     counts_df = (
         pl.scan_csv(nodes, separator="\t", truncate_ragged_lines=True, has_header=True, ignore_errors=True)
         .select(
             [
-                pl.col("id").str.contains(CURIE_REGEX).sum().alias("valid_curie_id_count"),
+                # we only really care about the invalids
+                # pl.col("id").str.contains(CURIE_REGEX).sum().alias("valid_curie_id_count"),
                 (~pl.col("id").str.contains(CURIE_REGEX)).sum().alias("invalid_curie_id_count"),
-                pl.col("category").str.contains(STARTS_WITH_BIOLINK_REGEX).sum().alias("valid_starts_with_biolink_category_count"),
+                # pl.col("id").str.contains_any(bm_prefixes).sum().alias("valid_contains_biolink_model_prefix_id_count"),
```

Comment:
Good call, it was kind of silly on my part to count both separately. Might as well remove the commented out lines?